### PR TITLE
Enable various p2sh-p2wpkh functionality

### PR DESCRIPTION
**Pull request by luke-jr**
A subset of #9017

### CODE REVIEW

File: src/base58.cpp [hunk 1]
```diff
@@ -281,6 +281,15 @@ bool CBitcoinAddress::GetKeyID(CKeyID& keyID) const
     return true;
 }
 
+bool CBitcoinAddress::GetScriptID(CScriptID& scriptID) const
+{
+    if (!IsValid() || vchVersion != Params().Base58Prefix(CChainParams::SCRIPT_ADDRESS)) {
+        return false;
+    }
+    memcpy(scriptID.begin(), &vchData[0], 20);
```
Comment (MeshCollider) [inline src/base58.cpp]: Prefer vchData.data() over &vchData[0]

**Comment by MeshCollider** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/11089/commits/7fb9887baa17fc0f0ea58bc891a1a25f88af9744

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/11089/commits/8f6765a697f9cfa193c5f6365012bc18e262c1fe

File: src/base58.cpp [hunk 2]
```diff
@@ -281,6 +281,15 @@ bool CBitcoinAddress::GetKeyID(CKeyID& keyID) const
     return true;
 }
 
+bool CBitcoinAddress::GetScriptID(CScriptID& scriptID) const
+{
+    if (!IsValid() || vchVersion != Params().Base58Prefix(CChainParams::SCRIPT_ADDRESS)) {
```
Comment (jimpo) [inline src/base58.cpp]: nit: Use `if (!IsScript())`

**Comment by laanwj** [comment]
Added 0.15.1 milestone

**Comment by luke-jr** [comment]
IMO it's a 0.16 thing.

**Comment by instagibbs** [comment]
needs rebase

**Comment by sipa** [comment]
This needs a rebase after CBitcoinAddress was removed in #11117. However, this will become much easier after #11167, as then native witness destinations can be used.

**Comment by MarcoFalke** [comment]
Needs rebase

**Comment by promag** [comment]
Either close (because it's in ##1403) or rebase?

**Comment by MarcoFalke** [comment]
Removing from backport

**Comment by luke-jr** [comment]
I don't know how to rebase this.

**Comment by sipa** [comment]
Isn't all the functionality in this PR merged through other patches? If something isn't, I'd like to know.

**Comment by instagibbs** [comment]
I believe so.

**Comment by WorkShop-Office** [comment]
Is there any bitcoin indonesia here

**Comment by WorkShop-Office** [comment]
dTurn me in. Closed there what ya with me do not so sacred

**Comment by luke-jr** [comment]
This had 4 changes:

* "pubkey" from DescribeAddressVisitor acting on a P2SH-wrapped Segwit address - This seems to be handled recursively or something now (although I don't see logic to prevent P2SH-within-P2SH, but that'd be an unrelated bug)
* validateaddress RPC gets metadata for 1-prefix addresses from witness addresses using it (if I'm reading the code correctly) - I don't see this in the 0.16 code, but it seems like a bad idea anyway.
* _createmultisig_redeemScript looked up full pubkey from Segwit addresses - Seems to be correctly handled in a very different way now.
* dumpprivkey from witness address - Seems to be correctly handled in a different way now.

**Pull request closed without merging**